lib/cmsis_rtos_v1/cmsis_thread.c

```diff
@@ -0,0 +1,246 @@
+
+static u32_t cmsis_to_zephyr_priority(s32_t c_prio)
+{
+	return(osPriorityRealtime - c_prio);
+}
+
+static void zephyr_thread_wrapper(void *arg1, void *arg2, void *arg3)
+{
+	osThreadId thread_id;
+	struct cmsis_thread *cm_thread;
+	void * (*fun_ptr)(void *) = arg3;
+
+	fun_ptr(arg1);
+
+	thread_id = osThreadGetId();
+
+	/* The below operation gets a pointer to the parent structure
+	 * of the thread which also has k_mem_block as one of its elements.
+	 * This needs to be used to free the mem pool.
+	 */
+	cm_thread = CONTAINER_OF(thread_id, struct cmsis_thread, thread);
+
+	worker.thrd = cm_thread;
+	k_work_submit(&worker.work.work);
+}
+
+/* Memory pool for cmsis thread space */
+K_MEM_POOL_DEFINE(cmsis_thread_pool, sizeof(struct cmsis_thread),
+		sizeof(struct cmsis_thread), CONFIG_MAX_CMSIS_THREAD_COUNT, 4);
+
+/**
+ * @brief Create a thread.
+ *
+ * Parameters
+ * [in]	thread_def - thread definition referenced with osThread.
+ * [in]	arg - pointer that is passed to the thread function as start argument.
+ *
+ * Returns
+ * thread ID for reference by other functions or NULL in case of error.
+ */
+osThreadId osThreadCreate(const osThreadDef_t *thread_def, void *arg)
+{
+	u32_t prio;
+	struct k_mem_block block;
+	struct cmsis_thread *cm_thread;
+	k_tid_t tid;
+
+	__ASSERT(thread_def->stacksize > 0 &&
+		 thread_def->stacksize <= CONFIG_MAX_STACK_SIZE_THREAD,
+		 "invalid stack size\n");
+
+	if (_is_in_isr()) {
+		return NULL;
+	}
+
+	if (!thread_def->instances) {
+		return NULL;
+	}
+
+	if (*(thread_def->used_inst_ptr) >= thread_def->instances) {
+		return NULL;
+	}
+
+	if (k_mem_pool_alloc(&cmsis_thread_pool, &block,
+			     sizeof(struct cmsis_thread), 100) != 0) {
+		/* Insufficient resource to create thread */
+		return NULL;
+	}
+
+	atomic_inc((atomic_t *)(thread_def->used_inst_ptr));
+	prio = cmsis_to_zephyr_priority(thread_def->tpriority);
+	memset(block.data, 0, sizeof(struct cmsis_thread));
+	cm_thread = block.data;
+
+	cm_thread->block.data = block.data;
+	cm_thread->block.id = block.id;
+
+	tid = k_thread_create(&cm_thread->thread,
+			thread_def->stack_mem[*(thread_def->used_inst_ptr) - 1],
+			thread_def->stacksize,
+			(k_thread_entry_t)zephyr_thread_wrapper,
+			(void *)arg, NULL, thread_def->pthread,
+			prio, 0, K_NO_WAIT);
+
+	return ((osThreadId)tid);
+}
+
+/**
+ * @brief Return the thread ID of the current running thread.
+ *
+ * Returns
+ * thread ID for reference by other functions or NULL in case of error.
+ */
+osThreadId osThreadGetId(void)
+{
+	if (_is_in_isr()) {
+		return NULL;
+	}
+
+	return (osThreadId)k_current_get();
+}
+
+/**
+ * @brief Get current priority of an active thread.
+ *
+ * Parameters
+ * [in] thread_id - thread ID obtained by osThreadCreate or osThreadGetId.
+ *
+ * Returns
+ * current priority value of the thread function.
+ */
+osPriority osThreadGetPriority(osThreadId thread_id)
+{
+	k_tid_t thread = (k_tid_t)thread_id;
+	u32_t priority;
+
+	if (_is_in_isr()) {
+		return osPriorityError;
+	}
```

Comment:
Again, seems like error checking isn't really appropriate.  Remove unless required by the spec.